packages/init/src/lib/framework.ts

```diff
@@ -58,11 +58,22 @@ export type Framework = {
 	devDependencies: Record<string, string>;
 };
 
-const isPrerelease =
-	semver.prerelease(pkgVersion) !== null && process.env.NODE_ENV !== "test";
+const parsedPkgVersion = semver.parse(pkgVersion);
+if (parsedPkgVersion === null) {
+	throw new Error(
+		`Package \`${pkgName}\` has an invalid version \`${pkgVersion}\` in its manifest.`,
+	);
+}
+let npmDistributionTag;
+if (parsedPkgVersion.prerelease.length > 0 && process.env.NODE_ENV !== "test") {
+	const prereleaseIdentifier = parsedPkgVersion.prerelease.slice(-1).join(".");
+	npmDistributionTag = prereleaseIdentifier;
+} else {
+	npmDistributionTag = "latest";
+}
 
 const DEFAULT_DEV_DEPENDENCIES: Record<string, string> = {
-	"slice-machine-ui": isPrerelease ? "alpha" : "latest",
+	"slice-machine-ui": npmDistributionTag,
```

Comment:
Just want to confirm: will this install dependencies using pinned versions (which is what we want)?

I.e. `npm install slice-machine-ui@0.1.2-alpha.my-branch.1` rather than `npm install slice-machine-ui@^0.1.2-alpha.my-branch.1`